storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -0,0 +1,3729 @@
+        return ret;
+    }
+    
+    private static void addToDecorators(Set<String> decorators, List<String> conf) {
+        if (conf != null) {
+            decorators.addAll(conf);
+        }
+    }
+    
+    @SuppressWarnings("unchecked")
+    private static void addToSerializers(Map<String, String> ser, List<Object> conf) {
+        if (conf != null) {
+            for (Object o: conf) {
+                if (o instanceof Map) {
+                    ser.putAll((Map<String,String>)o);
+                } else {
+                    ser.put((String)o, null);
+                }
+            }
+        }
+    }
+    
+    @SuppressWarnings("unchecked")
+    private static Map<String, Object> normalizeConf(Map<String,Object> conf, Map<String, Object> topoConf, StormTopology topology) {
+        //ensure that serializations are same for all tasks no matter what's on
+        // the supervisors. this also allows you to declare the serializations as a sequence
+        List<Map<String, Object>> allConfs = new ArrayList<>();
+        for (Object comp: StormCommon.allComponents(topology).values()) {
+            allConfs.add(StormCommon.componentConf(comp));
+        }
+
+        Set<String> decorators = new HashSet<>();
+        //Yes we are putting in a config that is not the same type we pulled out.
+        Map<String, String> serializers = new HashMap<>();
+        for (Map<String, Object> c: allConfs) {
+            addToDecorators(decorators, (List<String>) c.get(Config.TOPOLOGY_KRYO_DECORATORS));
+            addToSerializers(serializers, (List<Object>) c.get(Config.TOPOLOGY_KRYO_REGISTER));
+        }
+        addToDecorators(decorators, (List<String>)topoConf.getOrDefault(Config.TOPOLOGY_KRYO_DECORATORS, 
+                conf.get(Config.TOPOLOGY_KRYO_DECORATORS)));
+        addToSerializers(serializers, (List<Object>)topoConf.getOrDefault(Config.TOPOLOGY_KRYO_REGISTER, 
+                conf.get(Config.TOPOLOGY_KRYO_REGISTER)));
+        
+        Map<String, Object> mergedConf = merge(conf, topoConf);
+        Map<String, Object> ret = new HashMap<>(topoConf);
+        ret.put(Config.TOPOLOGY_KRYO_REGISTER, serializers);
+        ret.put(Config.TOPOLOGY_KRYO_DECORATORS, new ArrayList<>(decorators));
+        ret.put(Config.TOPOLOGY_ACKER_EXECUTORS, mergedConf.get(Config.TOPOLOGY_ACKER_EXECUTORS));
+        ret.put(Config.TOPOLOGY_EVENTLOGGER_EXECUTORS, mergedConf.get(Config.TOPOLOGY_EVENTLOGGER_EXECUTORS));
+        ret.put(Config.TOPOLOGY_MAX_TASK_PARALLELISM, mergedConf.get(Config.TOPOLOGY_MAX_TASK_PARALLELISM));
+        return ret;
+    }
+    
+    private static void rmBlobKey(BlobStore store, String key, IStormClusterState state) {
+        try {
+            store.deleteBlob(key, NIMBUS_SUBJECT);
+            if (store instanceof LocalFsBlobStore) {
+                state.removeBlobstoreKey(key);
+            }
+        } catch (Exception e) {
+            //Yes eat the exception
+            LOG.info("Exception {}", e);
```

Comment:
The "{}" will make it so the rsult of `e.toString()` is printed instead of the entire stack trace.